src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Runtime/LoggerMessageDefineAnalyzer.cs

```diff
@@ -0,0 +1,371 @@
+                                                                         DiagnosticCategory.Usage,
+                                                                         RuleLevel.IdeSuggestion,
+                                                                         description: s_localizableDescriptionCA2251,
+                                                                         isPortedFxCopRule: true,
+                                                                         isDataflowRule: false,
+                                                                         isReportedAtCompilationEnd: true);
+
+        internal static DiagnosticDescriptor CA2252Rule = DiagnosticDescriptorHelper.Create(CA2252RuleId,
+                                                                         s_localizableTitleCA2252,
+                                                                         s_localizableMessageCA2252,
+                                                                         DiagnosticCategory.Usage,
+                                                                         RuleLevel.BuildWarning,
+                                                                         description: s_localizableDescriptionCA2252,
+                                                                         isPortedFxCopRule: true,
+                                                                         isDataflowRule: false,
+                                                                         isReportedAtCompilationEnd: true);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(CA1727Rule, CA1839Rule, CA2250Rule, CA2251Rule, CA2252Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.ReportDiagnostics);
+
+            context.EnableConcurrentExecution();
+
+            context.RegisterCompilationStartAction(analysisContext =>
+            {
+                var wellKnownTypeProvider = WellKnownTypeProvider.GetOrCreate(analysisContext.Compilation);
+
+                if (!wellKnownTypeProvider.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.MicrosoftExtensionsLoggingLoggerExtensions, out var loggerExtensionsType) ||
+                    !wellKnownTypeProvider.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.MicrosoftExtensionsLoggingILogger, out var loggerType) ||
+                    !wellKnownTypeProvider.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.MicrosoftExtensionsLoggingLoggerMessage, out var loggerMessageType))
+                {
+                    return;
+                }
+
+                analysisContext.RegisterSyntaxNodeAction(syntaxContext => AnalyzeInvocation(syntaxContext, loggerType, loggerExtensionsType, loggerMessageType), SyntaxKind.InvocationExpression);
```

Comment:
It may be better to be written an IOperation analyzer if possible.

Reply:
I tried but it seems like it would be a big change to support it written with IOperation, and I'm not sure if it was possible. So kept as it, as was written for dotnet/extensions